vehicle_status_generator/src/vehicle_status_generator_worker.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2019-2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "vehicle_status_generator_worker.h"
+#include <stdlib.h>
+#include <ros/ros.h>
+#include <random>
+
+namespace vehicle_status_generator
+{
+    VehicleStatusGeneratorWorker::VehicleStatusGeneratorWorker() {}
+    
+    uint8_t VehicleStatusGeneratorWorker::getNextMsgCount()
+    {
+        uint8_t old_msg_count = msg_count_;
+        msg_count_++;
+        if(msg_count_ == 128)
+        {
+            msg_count_ = 0;
+        }
+        return old_msg_count;
+    }
+
+    std::vector<uint8_t> VehicleStatusGeneratorWorker::getMsgId(const ros::Time now)
```

Comment:
I'm not sure I understand the purpose of this method. I'm assuming this is related to the BSM ID field in the structure of the mobility operations message needed by CARMA Streets? If so, I'm not sure why we're generating a random ID here instead of actually reading the one we're currently broadcasting in the BSM? We have the BSM generator package already handling the generation and rotation of the BSM ID when it publishes the BSM for broadcast. We want to putting that data in the mobility message so that CARMA Streets can correlate the mobility message we broadcast at some timestep `t` with the BSM that was broadcast at time `t` as well.